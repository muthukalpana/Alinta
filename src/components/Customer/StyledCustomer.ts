import styled from "styled-components";
import { ScreenSize } from "../../enum/screenSize";

export const StyledCustomer = styled.div`
  box-sizing: border-box;
  @media (min-width: ${ ScreenSize.Mobile }) {
    flex: 0 0 90%;
  }
  @media (min-width: ${ ScreenSize.Tablet }) {
    flex: 0 0 46%;
  }
  @media (min-width: ${ ScreenSize.Laptop }) {
    flex: 0 0 30%;
  }
  @media (max-width > ${ ScreenSize.Desktop }) {
    flex: 0 0 20%;
  }
  border: 1px solid #ccc;
  margin: 1rem;
  padding: 1rem;
`;

export const StyledCustomerName = styled.h2`
  margin: 0 0 0.5rem;
  padding: 0;
`;

export const StyledCustomerInfo = styled.p`
  margin: 0 0 0.5rem;
  padding: 0;
`;

export const StyledCustomerDelete = styled.button`
  padding: 1rem;
  background-color: #dc1616;
  color: white;
  border: none;
  border-radius: 4px;
  font-size: 16px;
`;
