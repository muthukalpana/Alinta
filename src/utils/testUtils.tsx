import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { customerReducer } from "../redux/reducers/customerReducers";

// @TODO clean up types here
// Set-up copied from https://redux.js.org/recipes/writing-tests#components
function render(
  // @ts-ignore
  ui,
  {
    // @ts-ignore
    initialState,
    store = createStore(customerReducer, initialState),
    ...renderOptions
  } = {}
) {
  // @ts-ignore
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  // @ts-ignore
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
