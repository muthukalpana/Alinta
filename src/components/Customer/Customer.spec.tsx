import * as React from "react";
import { render } from "../../utils/testUtils";
import { Customer } from "./Customer";

const customer = {
  id: 1,
  firstName: "Test",
  lastName: "Tester",
  phoneNumber: "00000000",
};

describe("<Customer />", () => {
  it("should render a <Customer />", () => {
    const wrapper = render(
      <Customer customer={customer} removeCustomer={() => {}} />
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
