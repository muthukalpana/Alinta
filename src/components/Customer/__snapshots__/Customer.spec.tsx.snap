// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Customer /> should render a <Customer /> 1`] = `
<div>
  <div
    class="sc-bdfBQB dDOvEt"
  >
    <h2
      class="sc-gsTEea eRkGPM"
    >
      Test
       
      Tester
    </h2>
    <p
      class="sc-dlfnuX jIyJCw"
    >
      Phone number: 
      00000000
    </p>
    <button
      class="sc-hKgJUU dFuAzI"
    >
      Delete
    </button>
  </div>
</div>
`;
